src/openms/source/ANALYSIS/OPENSWATH/TargetedSpectraExtractor.cpp

```diff
@@ -162,21 +158,9 @@ namespace OpenMS
       for (UInt j=0; j<transitions.size(); ++j)
       {
         const TargetedExperimentHelper::Peptide peptide = targeted_exp.getPeptideByRef(transitions[j].getPeptideRef());
-        double target_rt;
-        if (!peptide.rts.empty() &&
-            peptide.rts[0].hasCVTerm("MS:1000895") &&
-            !peptide.rts[0].getCVTerms()["MS:1000895"].empty())
-        {
-          target_rt = peptide.rts[0].getCVTerms()["MS:1000895"][0].getValue().toString().toDouble();
-        }
-        else
-        {
-          target_rt = peptide.getRetentionTime(); // accesses to MS:1000896. TODO: update once https://github.com/OpenMS/OpenMS/pull/3043 is merged
-        }
-        target_rt *= (String)param_.getValue("rt_unit") == "seconds" ? 1.0 : 60.0;
-
+        double target_rt = peptide.getRetentionTime();
+        target_rt *= peptide.getRetentionTimeUnit() == TargetedExperimentHelper::RetentionTime::RTUnit::SECOND ? 1.0 : 60.0;
```

Comment:
I find:

if (peptide.getRetentionTimeUnit() == TargetedExperimentHelper::RetentionTime::RTUnit::MINUTES) { target_rt *= 60.0; }; 
a bit easier readable in this specific case

Reply:
but that's just personal taste

Reply:
That totally makes sense :+1: really no need to multiply something times 1 :sweat_smile: